module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -365,22 +339,28 @@ protected function buildIndex(array $sitemaps)
 
             // Add a <sitemap /> group for a static sitemap file.
             // See sitemap.ini for more information on this option.
-            $baseSitemapFileName = $this->config->SitemapIndex->baseSitemapFileName
-                ?? '';
-            if ($baseSitemapFileName) {
-                $baseSitemapFileName .= '.xml';
-                $baseSitemapFilePath = $this->fileLocation . '/'
-                    . $baseSitemapFileName;
-                // Only add the <sitemap /> group if the file exists
-                // in the directory where the other sitemap files
-                // are saved, i.e. ['Sitemap']['fileLocation']
-                if (file_exists($baseSitemapFilePath)) {
-                    $smf->addUrl($baseUrl . '/' . $baseSitemapFileName);
+            $indexSettings = $this->config->SitemapIndex->toArray();
+            $baseSitemapFileNames = (array)($indexSettings['baseSitemapFileName'] ?? []);
```

Comment:
Would this also provide a proper result if somebody still has a local copy of the sitemap.ini with the old non-array setting? (Not sure if supporting this would be necessary, maybe a changelog entry to migrate your old sitemap.ini would also be enough.)

Reply:
Yes, this should be backward compatible because of the `(array)` cast. If baseSitemapFileName is a single string, it will become an array containing that single string. If it's already an array, it will remain an array. Either way, the following loop will process it correctly.

Reply:
The test case covers both of these scenarios as confirmation. :-)